File: caesar.c
//~/pset2/caesar
//caesar.c
//pset2 cs50
//onelmv
//9.16.2018
//Caesar cypher

#include<stdio.h>
#include<cs50.h>
#include<stdlib.h>
#include<string.h>
#include<ctype.h>

char Encrypting(char c, int key);

int main(int argc, string argv[])
{
    int key = 0;

    if (argc <= 2 && argc > 1) // Run if there are only two parameter.
    {
        key = atoi(argv[1]);

        if (key > 0) 
        {
            string plain_text = get_string("plaintext: "); 
            int len_text = strlen(plain_text);  

            printf("ciphertext: ");
            for (int i = 0 ; i < len_text; i++)
            {
                printf("%c", Encrypting(plain_text[i], key));
            }
            printf("\n");
        }
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}//end

char Encrypting(char c, int key)
{
    if (isalpha(c))
    {
        if (isupper(c))
        {
            return (c - 65 + key) % 26 + 65;

        }
        else
        {   
            //islower()
            return (c - 97 + key) % 26 + 97;
        }
    }
    //no change
    return c;
}

File: caesar.py
# ~/pset6/caesar
# caesar.py
# pset6 cs50
# onelmv
# 12.26.2018
# Caesar cypher

import sys
from cs50 import get_string

# cheking the argument exist
if len(sys.argv) == 2:

    # checking if the argument it's a number
    if sys.argv[1].isdigit():

        # converting the key (str -> int)
        key = int((sys.argv[1]))

        # Taking text from the user
        text = get_string("plaintext: ")

        # print tag
        print("ciphertext: ", end="")
        for i in text:
            # encripting only the alpha character
            if i.isalpha():
                # Checking if the char is Upper or lower case
                if i.isupper():
                    # encrypting and printing
                    # ord() convert to int, and chr convert to str again
                    print(chr((ord(i) - 65 + key) % 26 + 65), end="")
                else:  # i.islower()
                    # encrypting and printing
                    print(chr((ord(i) - 97 + key) % 26 + 97), end="")
            #  non-alphabetical characters should be outputted unchanged.
            else:
                print(i, end="")
        print()
    else:
        print("Usage: ./caesar key")
else:
    print("Usage: ./caesar key")
    exit(1)